### PR TITLE
Publish test package at AGENT_WORKFOLDER if TEMP is not available

### PR DESCRIPTION
**Pull request by adityapatwardhan**
## PR Summary

macOS hosted agents on Azure DevOps do not have `$env:TEMP` set. In that case, use `$env:AGENT_WORKFOLDER`

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: build.psm1 [hunk 1]
```diff
@@ -2997,6 +2997,12 @@ function New-TestPackage
         Write-Verbose -Message "Creating destination folder: $Destination"
     }
 
+    $rootFolder = $env:TEMP
+
+    if (-not $rootFolder -and $env:TF_BUILD) {
+        $rootFolder = $env:AGENT_WORKFOLDER
+    }
```
Comment (JamesWTruher) [inline build.psm1]: should there be an `else` here, just in case?

**Review by JamesWTruher — APPROVED** [review]
just one comment, and not a blocker in this case

File: build.psm1 (hunk 1 above)
Comment (adityapatwardhan) [reply build.psm1]: First it tries to set root folder to $env:temp if it is $null then it tries to set it to AGENT_WOORKFOLDER only if running under VSTS (checked by $env:TF_BUILD)

File: build.psm1 [hunk 2]
```diff
@@ -2997,7 +2997,13 @@ function New-TestPackage
...
     }
 
-    $packageRoot = Join-Path $env:TEMP ('TestPackage-' + (new-guid))
+    $rootFolder = $env:TEMP
+
+    if (-not $rootFolder -and $env:TF_BUILD) {
+        $rootFolder = $env:AGENT_WORKFOLDER
+    }
+
+    $packageRoot = Join-Path $rootFolder ('TestPackage-' + (new-guid))
```
Comment (PaulHigin) [inline build.psm1]: Is it possible to have both `$env:Temp` and `$env:TF_BUILD` be null?  Maybe we should have a `Test-Path` here and throw an error message ... "package path cannot be found"?

**Review by PaulHigin — APPROVED** [review]

File: build.psm1 (hunk 2 above)
Comment (adityapatwardhan) [reply build.psm1]: $env:TEMP should be set in most cases. VSTS macOS hosts seem to be the only place where it is not set. $env:TF_BUILD is always set in VSTS builds.

**Pull request merged**